compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -589,6 +590,23 @@ private void addToCompiledSymbols(Set<Symbol> compiledSymbols, Scope.ScopeEntry
         addToCompiledSymbols(compiledSymbols, scopeEntry.next, cursorPos, name, symbolEnv, states, compUnitName);
     }
 
+    private boolean checkAndUpdateModuleSymbols(Set<Symbol> compiledSymbols, Symbol evaluatingSymbol, BSymbol symbol) {
+        boolean isContaining = compiledSymbols.contains(evaluatingSymbol);
+
+        if (evaluatingSymbol.kind() != MODULE) {
+            return isContaining;
+        }
+
+        // If the same module symbol, but without a module alias is already added, then it shall be removed to add
+        // the new symbol with the import alias.
+        if (((BPackageSymbol) symbol).importPrefix != null) {
+            compiledSymbols.remove(evaluatingSymbol);
```

Comment:
For pre-declared imports, this would be an issue.

```ballerina

import ballerina/lang.'int as foo;

public function main() {
    _ = int:max(20, 10);
}
```

Reply:
Updated the approach with an early-return if not-existing in https://github.com/ballerina-platform/ballerina-lang/pull/38093/commits/e228debdfd2b6ee4c5d3c2ee4750aedf511ea062.